cmds/fsdb.c

```diff
@@ -0,0 +1,176 @@
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <time.h>
+#include <linux/fs.h>
+#include <linux/types.h>
+#include <stdlib.h>
+#include <string.h>
+#include "../mfs.h"
+
+
+struct mfs_super_block sb;
+struct mfs_inode_map imap;
+struct mfs_block_map bmap;
+int devfd;
+
+int read_inode(int block_no, int offset, struct mfs_inode *uip)
+{	
+//	printf("imap.map[2] : %d \n",imap.map[2]);
+
+	lseek(devfd,((block_no*MFS_BLOCKSIZE)+offset),SEEK_SET);
+	read(devfd, (char *)uip, sizeof(struct mfs_inode));
+	return 0;
+}
+
+int read_sbnode(int block_no, int offset,struct mfs_super_block *usp)
+{
+	lseek(devfd,((block_no*MFS_BLOCKSIZE)+offset),SEEK_SET);
+	read(devfd, (char *)usp, sizeof(struct mfs_super_block));
+	return 0;
+}
+
+void print_sbnode(struct mfs_super_block *usp)
+{
+	printf("msb_magic                :0x%x\n",usp->msb_magic);
+        printf("msb_n_free_inode         : %d \n",usp->msb_n_free_inode);
+        printf("msb_n_free_blks          : %d \n",usp->msb_n_free_blks);
+
+}
+
+void print_inode(struct mfs_inode *uip)
+{
+	char buf[MFS_BLOCKSIZE];
+	struct mfs_directory_entry *dirent;
+	int i,x;
+
+
+//	printf("\nInode numner : %d \n",inum);
+	printf("mi_mode            : %x\n",uip->mi_mode);
+	printf("mi_uid             : %d\n",uip->mi_uid);
+	printf("mi_gid             : %d\n",uip->mi_gid);
+	printf("mi_atime           : %s\n",ctime((time_t *)&uip->mi_atime));
+	printf("mi_ctime           : %s\n",ctime((time_t *)&uip->mi_ctime));
+	printf("mi_mtime           : %s\n",ctime((time_t *)&uip->mi_mtime));
+	printf("mi_links_count     : %d\n",uip->mi_links_count);
+	printf("mi_size            : %d\n",uip->mi_size);
+	printf("mi_blocks          : %d\n",uip->mi_blocks);
+	
+	for(i=0;i<MFS_IBLOCK_COUNT;i++)
+	{
+		if(i%4==0)
```

Comment:
I will prefer to print a table like this

block number :  1          2
address           :   4213   92340